packages/components/stamp/src/stamp.tsx

```diff
@@ -133,16 +150,42 @@ const getStampStyles = (props: StylesFunctionParams) => {
 const Stamp = (props: Props) => {
   const { themedValue } = useTheme();
   const overrideTextColor = themedValue(false, true);
+  const Icon =
+    props.icon &&
+    cloneElement(props.icon, {
+      size: 'medium',
+      color: props.tone ? iconColorsMap[props.tone] : null,
+    });
 
-  return (
+  return props.children ? (
```

Comment:
I would give priority to the new use case. Also, perhaps it would more clear to not use a ternary here:
```jsx
if (props.message) {
  return (....);
}

// use children
return (....)
```

Reply:
Thank you. Updated here: https://github.com/commercetools/ui-kit/pull/2383/commits/f9e56c683f1769bbd086e070a943603014706791